### DISCUSSION TITLE
How to generate this Typescript keyof union

### DISCUSSION THREAD

**Discussion by djMax** [babel/babel#15195] [Q&A]
I'm not able to figure out how to generate this type:

```
const Type = string[] | keyof someObject;
```

This is the context, but I cannot figure out how to invoke this type operator with @babel/types.

```
          T.unionTypeAnnotation([
            T.arrayTypeAnnotation(T.stringTypeAnnotation()),
            // TODO make this more specific
            T.stringTypeAnnotation(),
          ]),
```

Thanks!

**Comment by liuxingbaoyu** [comment]
https://www.typescriptlang.org/play?noImplicitAny=false&ts=4.8.0-beta#code/MYewdgzgLgBAKgTwA4FMYF4bQE4EswDmA2gLowA+MA1igiAGZYgC2KA8gEYBWKwUA3EA

This is illegal.

**Comment by liuxingbaoyu** [comment]
https://astexplorer.net/#/gist/6e844207e8176f28aba7acbbc08d541d/833b641b64b88ce3d2bdc4d9eea9737f5a5a14f0

If you mean `type Type = string[] | keyof someObject;` you can refer to this.
